### PR TITLE
feat: Hooks and API simplifications

### PR DESCRIPTION
**Pull request by jquense**

### CODE REVIEW

File: src/DropdownMenu.js [hunk 1]
```diff
@@ -109,23 +98,23 @@ class DropdownMenu extends React.Component {
...
-    } = this.props;
+    const { flip, usePopper, rootCloseEvent, popperConfig = {} } = this.props;
+
+    const { drop, menuRef, toggleNode } = this.context;
+
+    const show =
+      this.context.show == null ? this.props.show : this.context.show;
+
+    const alignEnd =
+      this.context.alignEnd == null
```
Comment (taion) [inline src/DropdownMenu.js]: shouldn't props win over context in general?

File: src/DropdownMenu.js [hunk 2]
```diff
@@ -188,17 +177,13 @@ class DropdownMenu extends React.Component {
...
-    toggleNode,
-    onToggle: toggle,
-    show: show == null ? props.show : show,
-    alignEnd: alignEnd == null ? props.alignEnd : alignEnd,
-  }),
-  DropdownMenu,
-);
-
-export default DecoratedDropdownMenu;
+const f = ({ show, alignEnd, toggle, drop, menuRef, toggleNode }, props) => ({
```
Comment (taion) [inline src/DropdownMenu.js]: we should be exporting this `f`? or at least using it?

Comment (taion) [reply src/DropdownMenu.js]: or maybe we're not using this at all?

File: src/DropdownToggle.js [hunk 3]
```diff
@@ -21,22 +33,15 @@ const propTypes = {
    */
   children: PropTypes.func.isRequired,
 };
+
 function DropdownToggle({ children }) {
```
Comment (taion) [inline src/DropdownToggle.js]: is this essentially soft-deprecated in favor of the hook now?

File: src/Overlay.js [hunk 4]
```diff
@@ -1,166 +1,126 @@
...
+  const popperProps = {
+    ...popperConfig,
+    placement,
+    innerRef: ref,
+    referenceElement: target,
+    enableEvents: props.show,
+    modifiers: {
+      ...modifiers,
+      preventOverflow: {
+        padding: containerPadding || 5,
```
Comment (taion) [inline src/Overlay.js]: can we use `defaultProps` here?

File: www/gatsby-config.js [hunk 5]
```diff
@@ -21,7 +21,7 @@ module.exports = {
         name: 'source',
       },
     },
-    'gatsby-plugin-emotion',
+    // 'gatsby-plugin-emotion',
```
Comment (taion) [inline www/gatsby-config.js]: ?

File: src/DropdownMenu.js (hunk 2 above)
Comment (jquense) [reply src/DropdownMenu.js]: just left over while refactoring oops

File: src/DropdownToggle.js (hunk 3 above)
Comment (jquense) [reply src/DropdownToggle.js]: maybe, i'm not convinced the pure hook api here is the best option, at least not until there is a proper popper.js hooks api

File: src/DropdownMenu.js (hunk 1 above)
Comment (jquense) [reply src/DropdownMenu.js]: probably...this was the way it was i think

File: www/gatsby-config.js (hunk 5 above)
Comment (jquense) [reply www/gatsby-config.js]: it doesn't work :/

File: src/RootCloseWrapper.js [hunk 6]
```diff
@@ -1,20 +1,89 @@
...
+  { disabled, clickTrigger = 'click' } = {},
+) {
+  const preventMouseRootCloseRef = useRef(false);
+  const onClose = useEventCallback(onRootClose || noop);
+
+  const handleMouseCapture = e => {
+    const currentTarget = ref && ('current' in ref ? ref.current : ref);
+    warning(
+      !!currentTarget,
+      'RootClose captured a close event but does not have a ref to compare it to. ' +
```
Comment (taion) [inline src/RootCloseWrapper.js]: i'd just pre-concat these; now that we're on prettier, we won't get a line length lint hit anyway

File: src/RootCloseWrapper.js [hunk 7]
```diff
@@ -1,20 +1,89 @@
...
+    // avoid false positives in the contains() check below if the target DOM
+    // element is removed in the React mouse callback.
+    const removeMouseCaptureListener = listen(
+      document,
+      clickTrigger,
+      handleMouseCapture,
+      true,
+    );
+
+    const removeMouseListener = listen(document, clickTrigger, handleMouse);
```
Comment (taion) [inline src/RootCloseWrapper.js]: do we not need `handleMouse` and `handleKeyUp` for exhaustive dep tracking?

File: src/utils/mergeRefs.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import { useMemo } from 'react';
```
Comment (taion) [inline src/utils/mergeRefs.js]: move to restart/hooks?

**Comment by jquense** [comment]
gonna clean up the lint in a min

File: package.json [hunk 9]
```diff
@@ -51,38 +51,44 @@
...
-    "react-context-toolbox": "^2.0.2",
-    "react-popper": "^1.3.2",
-    "uncontrollable": "^6.0.0",
-    "warning": "^4.0.2"
+    "react-popper": "^1.3.3",
+    "uncontrollable": "^6.1.0",
+    "warning": "^4.0.3"
   },
   "peerDependencies": {
     "react": ">=16.3.0",
```
Comment (nihgwu) [inline package.json]: we should bump up the peer dependencies as we are using hooks now

**Pull request merged**